src/hotspot/cpu/aarch64/atomic_aarch64.S

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2021, Red Hat Inc. All rights reserved.
+// DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+
+// This code is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License version 2 only, as
+// published by the Free Software Foundation.
+
+// This code is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// version 2 for more details (a copy is included in the LICENSE file that
+// accompanied this code).
+
+// You should have received a copy of the GNU General Public License version
+// 2 along with this work; if not, write to the Free Software Foundation,
+// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+
+// Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+// or visit www.oracle.com if you need additional information or have any
+// questions.
+
+
+
+        .text
+
+        .globl aarch64_atomic_fetch_add_8_default_impl
+aarch64_atomic_fetch_add_8_default_impl:
+0:      ldaxr   x2, [x0]
+        add     x8, x2, x1
+        stlxr   w9, x8, [x0]
+        cbnz    w9, 0b
+        mov     x0, x2
+        ret
+
+        .globl aarch64_atomic_fetch_add_4_default_impl
```

Comment:
The N1 optimisation guide suggests aligning branch targets on 32 byte boundaries.

Reply:
OK.